dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,441 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+
+        try {
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", testCase.defaultContentToDefaultLanguage);
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", testCase.defaultPageToDefaultLanguage);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
+            folder = new FolderDataGen().site(site).nextPersisted();
+
+            Contentlet contentletDefaultLang = null;
+            Contentlet contentSecondLang = null;
+
+            if(testCase.createContentInDefaultLanguage) {
+                contentletDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
+
+                if(testCase.createContentInSecondLanguage) {
+                    createNewVersionAndPublishExistingEmployeeContent(secondLang, contentletDefaultLang,
+                            "catalina");
+                }
+            } else if(testCase.createContentInSecondLanguage) {
+                contentSecondLang = createAndPublishEmployeeContent(site, secondLang, "catalina");
+            }
+
+            HTMLPageAsset pageDefaultLang = null;
+            HTMLPageAsset pageSecondLang = null;
+
+            Contentlet contentToPassToPage = contentletDefaultLang!=null
+                    ? contentletDefaultLang
+                    : contentSecondLang;
+
+            if(testCase.createPageInDefaultLanguage) {
+                pageDefaultLang = createHtmlPageAsset(defaultLang, contentToPassToPage);
+
+                if(testCase.createPageInSecondLanguage) {
+                    pageSecondLang = createNewVersionHtmlPage(pageDefaultLang, secondLang);
+                }
+            } else if(testCase.createPageInSecondLanguage) {
+                pageSecondLang = createHtmlPageAsset(secondLang, contentToPassToPage);
+            }
+
+            final List<String> indicesBeforeTest = siteSearchAPI.listIndices();
+            for (final String index : indicesBeforeTest) {
+                esIndexAPI.delete(index);
+            }
+            final String jobId = UUIDUtil.uuid();
+            final String alias = "any-alias-" + System.currentTimeMillis();
+            final JobDataMap jobDataMap = new JobDataMap();
+            jobDataMap.put(SiteSearchJobImpl.RUN_NOW, Boolean.TRUE.toString());
+            jobDataMap.put(SiteSearchJobImpl.INCREMENTAL, Boolean.FALSE.toString());
+            jobDataMap.put(SiteSearchJobImpl.INDEX_ALIAS, alias);
+            jobDataMap.put(SiteSearchJobImpl.JOB_ID, jobId);
+            jobDataMap.put(SiteSearchJobImpl.QUARTZ_JOB_NAME,
+                    SiteSearchJobImpl.RUNNING_ONCE_JOB_NAME);
+            jobDataMap.put(SiteSearchJobImpl.INCLUDE_EXCLUDE, "all");
+
+            List<String> langsToIndex = new ArrayList<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'langsToIndex' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)